Failing DW test on executorch

skrtskrtfb opened the pull request:
Summary: Example of Failing DW Test on ARM U55 (set quantize_io = True)

Specific configurations of depthwise convolutions fail in the ethosu/vela/tosa_graph_optimizer, citing inability to schedule the DW conv on the NPU. On further investigation, it looks like the incoming tensor dimension gets permuted incorrectly, and the batch dimension/channel dimensions get swapped.

Example failing test in PR (dw_conv2d_3x3_1x3x256x256_gp3_st1), would appreciate help/direction with regards to a solution.

Differential Revision: D61862425

digantdesai commented:
@freddan80 / @robell - can someone help look into this? @skrtskrtfb described this findings and a potential smoking gun in the summary

Erik-Lundell commented:
Hi again, after some digging I am quite sure that the issue lies in `the is_weight_node_for_depthwise_conv2d` function in `annotate_channels_last_dim_order_pass.py` The function is supposed to only return True for depthwise_conv2d weights but has a false positive for the input `quantized_decomposed_quantize_per_tensor_default`. This causes the pass to add incorrect dim_order meta to the input tensor which down the line means an incorrect permutation as you mentioned.

I'll try to find some time to address it but that's somewhere to start if you'd like to give it a go yourself. Hope it helps!
/Erik

skrtskrtfb commented:
Thanks @Erik-Lundell, looks like that was the error after all! Just pushed a patch to resolve the issue

Erik-Lundell commented:
Hi again skrtskrtfb! I can't locate your patch, is it on its way?

skrtskrtfb commented:
@Erik-Lundell apologies, did not realize it didn't get pushed to the github repo - should be there now!

Commit:
Failing DW test on executorch (#4929)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4929

example of failing U55 DW test

Reviewed By: hsharma35

Differential Revision: D61862425

zingo commented:
Thanks!
